core/exception/system_call_error_spec.rb

```diff
@@ -20,41 +20,82 @@ def initialize
 end
 
 describe "SystemCallError.new" do
+  before :all do
+    # EINVAL/22 should be consistent across platforms.
+    @example_errno = 22
+    @example_errno_class = Object.const_get("Errno::EINVAL")
+    @last_known_errno = Errno.constants.size - 1
+    @unknown_errno = Errno.constants.size
+  end
+
   it "requires at least one argument" do
     -> { SystemCallError.new }.should raise_error(ArgumentError)
   end
 
   it "accepts single Fixnum argument as errno" do
     SystemCallError.new(-2**24).errno.should == -2**24
-    SystemCallError.new(42).errno.should == 42
+    SystemCallError.new(-1).errno.should == -1
+    SystemCallError.new(0).errno.should == 0
+    SystemCallError.new(@last_known_errno).errno.should == @last_known_errno
+    SystemCallError.new(@unknown_errno).errno.should == @unknown_errno
     SystemCallError.new(2**24).errno.should == 2**24
   end
 
+  it "constructs a SystemCallError for an unknown error number" do
+    SystemCallError.new(-2**24).should be_an_instance_of(SystemCallError)
+    SystemCallError.new(-1).should be_an_instance_of(SystemCallError)
+    SystemCallError.new(@unknown_errno).should be_an_instance_of(SystemCallError)
+    SystemCallError.new(2**24).should be_an_instance_of(SystemCallError)
+  end
   it "constructs the appropriate Errno class" do
-    # EINVAL should be more or less mortable across the platforms,
-    # so let's use it then.
-    SystemCallError.new(22).should be_kind_of(SystemCallError)
-    SystemCallError.new(22).should be_an_instance_of(Errno::EINVAL)
-    SystemCallError.new(2**28).should be_an_instance_of(SystemCallError)
+    e = SystemCallError.new(@example_errno)
+    e.should be_kind_of(SystemCallError)
+    e.should be_an_instance_of(@example_errno_class)
   end
 
   it "accepts an optional custom message preceding the errno" do
-    exc = SystemCallError.new("custom message", 22)
-    exc.should be_an_instance_of(Errno::EINVAL)
-    exc.errno.should == 22
-    exc.message.should == "Invalid argument - custom message"
```

Comment:
I think this check was useful, and clearer about what the full message should be, could you keep it?

Reply:
Yes, I agree, and will restore.

Earlier I had tried to select an example name/num from Errno.constants, and so had removed the specific msg part here.
I was surprised to find that Errno.constants.index(:EINVAL) is not 22 in either my Win or Linux OS!  Apparently the 22 is computed or looked up here.